File: ruby/bob/bob.rb
class Bob
  attr_reader :remark

  def hey(remark)
    @remark = remark.gsub(/\s/, '')
    return 'Fine. Be that way!' if silent?
    return 'Whoa, chill out!'   if shouting? & letters?
    return 'Sure.'              if asking?
    'Whatever.'
  end

  private

  def silent?
    remark == ''
  end

  def shouting?
    remark.upcase == remark
  end

  def letters?
    (remark =~ /[a-zA-Z]/)
  end

  def asking?
    remark =~ /\?$/
  end
end

File: elixir/bob/bob.exs
defmodule Bob do
  def hey(remark) do
    cond do
      !Regex.match?(~r/\w/, remark)    -> "Fine. Be that way!"
      String.ends_with?(remark, "?")   -> "Sure."
      remark == String.upcase(remark)  && Regex.match?(~r/[A-ZД]/, remark) -> "Whoa, chill out!"
      true -> "Whatever."
    end
  end
end
